recipe/meta.yaml

```diff
@@ -40,22 +46,19 @@ outputs:
       build:
         - {{ compiler("c") }}
         - {{ compiler("cxx") }}
-        - arm-variant * {{ arm_variant_type }}  # [aarch64]
-        - sysroot_{{ target_platform }} 2.17    # [linux]
-      host:
-        - cuda-version {{ cuda_version }}
-        - cuda-cudart-dev
+        - arm-variant * {{ arm_variant_type }}        # [aarch64]
+        - sysroot_{{ target_platform }} 2.17          # [linux]
       run:
-        - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - binutils               # [linux]
+        - gcc                    # [linux]
+        - gxx                    # [linux]
```

Comment:
To clarify the intent: how do `{{ compiler("c") }}` and `gcc` differ here? This is in an `if` block that requires `target_platform == cross_target_platform == cross_target_impl_platform`, so I expected those to be the same platform. Assuming that the platform requirement acts as I said, I expect the dependency resolution to work like this:

For `linux-64`:
- `{{ compiler("c") }}` is the same as `gcc_linux-64`. [Its dependencies](https://anaconda.org/conda-forge/gcc_linux-64/files) are:
  - `binutils_linux-64, gcc_impl_linux-64, sysroot_linux-64`
- `gcc`'s [dependencies](https://anaconda.org/conda-forge/gcc/files) are:
  - `gcc_impl_linux-64`. Only one dep, and it's already included by `{{ compiler("c") }}` above. So this is redundant?

Am I understanding the behavior correctly here?

Reply:
These are taken from [`c-compiler`]( https://github.com/conda-forge/compilers-feedstock/blob/c4865f325ba752842f7ea21470abde447f7f1062/recipe/meta.yaml#L40 ) and [`cxx-compiler`]( https://github.com/conda-forge/compilers-feedstock/blob/c4865f325ba752842f7ea21470abde447f7f1062/recipe/meta.yaml#L69 ). Maybe we could also consider just depending on those directly

Reply:
Let's go ahead and replace these with the corresponding `*-compiler` packages to simplify this. Including the suggested change below

Reply:
Done in commit ( https://github.com/conda-forge/cuda-nvcc-feedstock/commit/b2dc9ecfe2ff08622fd103691f4706e7edd2ca30 )